code/modules/projectiles/projectile/energy.dm

```diff
@@ -20,7 +24,28 @@
 	//Damage will be handled on the MOB side, to prevent window shattering.
 
 /obj/item/projectile/energy/electrode/on_hit(var/atom/target, var/blocked = 0)
+	var/deflected = NO_DEFLECTION //0 for not deflected, 1 for partially, 2 for totally.
+	if(ishuman(target)) //have to do this stuff before we call ..(), but I want the messages to come in the right order. So the deflected var.

```

Comment:
This is all gross. You're defining behavior for an entire armor on a single projectile. This is inconsistent with how all other armor handling is done (one the mob).

*If* This PR's idea was hypothetically accepted, and this system was to expand, then someone has to copy-paste this behavior over to other projectiles--that's not a good thing.